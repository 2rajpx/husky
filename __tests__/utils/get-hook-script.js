const getHookScript = require('../../src/utils/get-hook-script')

describe('getHookScript', () => {
  it('should render', () => {
    expect(getHookScript('pre-commit', '.', 'precommit')).toMatchSnapshot()
  })
})
