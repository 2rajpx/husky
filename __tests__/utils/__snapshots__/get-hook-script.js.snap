// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHookScript should render 1`] = `
"#!/bin/sh
#husky 0.13.4

command_exists () {
  command -v \\"$1\\" >/dev/null 2>&1
}

# https://github.com/typicode/husky/issues/76
has_hook_script () {
  [ -f package.json ] && cat package.json | grep -q \\"\\\\\\"$1\\\\\\"[[:space:]]*:\\"'
}

cd .

# Fix for issue #16 #24
# If script is not defined in package.json then exit
has_hook_script precommit || exit 0

load_nvm () {
  export $1=$2
  command_exists nvm || [ -s \\"$2/nvm.sh\\" ] && . $2/nvm.sh
  command_exists nvm && [ -f .nvmrc ] && nvm use
}

export PATH=$PATH:/usr/local/bin:/usr/local
      
load_nvm BREW_NVM_DIR /usr/local/opt/nvm
load_nvm NVM_DIR /Users/perso/.nvm
# Test if npm is in PATH
command_exists npm || {,
  echo >&2 \\"> husky - Can't find npm in PATH. Skipping precommit script in package.json\\",
  exit 0
}

# Expose GIT params
export GIT_PARAMS=\\"$*\\"

# Run script
echo
echo \\"> husky - npm run -s precommit\\"
echo \\"> husky - node \`node -v\`\\"'
echo
npm run -s precommit || {
  echo
  echo \\"> husky - pre-commit hook failed (add --no-verify to bypass)\\"
  echo \\"> husky - to debug, use 'npm run precommit'\\"
  exit 1
}"
`;
